go/anonymous-race-condition.yaml

```diff
@@ -59,58 +69,28 @@ rules:
             ...
           }
     - pattern-not: |
-        for $X, $Y := range ... {
-          ...
-          go func(..., $V, ...){
-            ...
-            $FOO(..., $V, ...)
-            ...
-          }(..., $Y, ...)
-          ...
-        }
-    - pattern-not: |
-        for $Y, $X := range ... {
-          ...
-          go func(..., $CP, ...){
-            ...
-            $CP(...)
-            ...
-          }($X)
-          ...
-        }
-    - pattern-not: |
-        for $X:=...;$Y;$Z {
-          ...
-          go func(..., $V, ...) {
-            ...
-            $FOO(..., $V,...)
-            ...
-          }(..., $X, ...)
-          ...
-        }
-    - pattern-not: |
-        for $X, $Y := range ... {
+        for ..., $X := range ... {
           ...
-          $COPY := $Y
+          ..., $X := ..., $X
           ...
           go func(...){
             ...
-            $FOO(..., $COPY, ...)
+            $FOO(..., $X, ...)
             ...
           }(...)
           ...
         }
     - pattern-not: |
-        for $X:=...;$Y;$Z {
-          ...
-          $COPY := $X
-          ...
-          go func(...) {
+        for ..., $X := range ... {
             ...
-            $FOO(..., $COPY,...)
+            $X, ... := $X, ...
+            ...
+            go func(...){
+              ...
+              $FOO(..., $X, ...)
```

Comment:
Do both these `pattern-not`s need to exist? Or could the body of the `go func` just be `...`?
```yaml
    - pattern-not: |
        for $Y, $X := range ... {
          ...
          $Y, $X := $Y, $X
          ...
          go func(...){
            ...
            <... $X.$VAR ...>
          }(...)
          ...
        }
    - pattern-not: |
        for $Y, $X := range ... {
          ...
          $Y, $X := $Y, $X
          ...
          go func(...){
            ...
            <... $Y.$VAR ...>
          }(...)
          ...
        }
```

I think we can replace these two with just:
```
    - pattern-not: |
        for $Y, $X := range ... {
          ...
          $Y, $X := $Y, $X
          ...
          go func(...){
            ...
          }(...)
          ...
        }
```

Reply:
> Do both these `pattern-not`s need to exist? Or could the body of the `go func` just be `...`?
> 
> ```yaml
>     - pattern-not: |
>         for $Y, $X := range ... {
>           ...
>           $Y, $X := $Y, $X
>           ...
>           go func(...){
>             ...
>             <... $X.$VAR ...>
>           }(...)
>           ...
>         }
>     - pattern-not: |
>         for $Y, $X := range ... {
>           ...
>           $Y, $X := $Y, $X
>           ...
>           go func(...){
>             ...
>             <... $Y.$VAR ...>
>           }(...)
>           ...
>         }
> ```
> 
> I think we can replace these two with just:
> 
> ```
>     - pattern-not: |
>         for $Y, $X := range ... {
>           ...
>           $Y, $X := $Y, $X
>           ...
>           go func(...){
>             ...
>           }(...)
>           ...
>         }
> ```

Agreed.  I will make the update to that

Reply:
done